lib/vscode/src/vs/platform/extensionManagement/common/extensionGalleryService.ts

```diff
@@ -363,8 +363,13 @@ export class ExtensionGalleryService implements IExtensionGalleryService {
 		this.commonHeadersPromise = resolveMarketplaceHeaders(productService.version, this.environmentService, this.fileService, storageService);
 	}
 
-	private api(path = ''): string {
-		return `${this.extensionsGalleryUrl}${path}`;
+	private api(path = ''): string[] {
+		if (!!this.extensionsGalleryUrl) {
```

Comment:
Is the double `!` to convert it to a boolean? Does it have a performance advantage over the [Boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) function? Only asking/suggesting if it might improve readability - obviously your call :D 
```suggestion
		if (Boolean(this.extensionsGalleryUrl)) {
```

Reply:
Perhaps I should just use `this.isEnabled()` instead - does the same check in practice but is probably ideal since the code already exists.

Reply:
Good thinking!